Source/SA/DPD/DPD.cs

```diff
@@ -0,0 +1,207 @@
+        {
+            LookupTableResults lutResults = new LookupTableResults();
+            IntPtr rfsgPtr;
+            RFmxSpecAnMXDpdApplyDpdIdleDurationPresent idlePresent;
+            idlePresent = waveform.IdleDurationPresent ? RFmxSpecAnMXDpdApplyDpdIdleDurationPresent.True : RFmxSpecAnMXDpdApplyDpdIdleDurationPresent.False;
+            rfsgPtr = rfsgSession.GetInstrumentHandle().DangerousGetHandle();
+            specAnSignal.Initiate("", selectorString);
+            specAnSignal.Dpd.ApplyDpd.ConfigureLookupTableCorrectionType(selectorString, lutConfig.LookupTableCorrectionType);
+            specAnSignal.Dpd.ApplyDpd.ApplyDigitalPredistortion(selectorString, waveform.WaveformData, idlePresent, commonConfig.Timeout, ref lutResults.WaveformWithDpdComplexSingle,
+                out lutResults.PostDPDPAPR, out lutResults.PowerOffset);
+
+            waveform.WaveformData = lutResults.WaveformWithDpdComplexSingle;
+            waveform.SampleRate = 1 / lutResults.WaveformWithDpdComplexSingle.PrecisionTiming.SampleInterval.TotalSeconds;
+            waveform.PAPR_dB = lutResults.PostDPDPAPR;
+            waveform.SignalBandwidth_Hz = 0.8 * waveform.SampleRate;
+
+            SG.DownloadWaveform(rfsgSession, waveform);
+            rfsgSession.RF.PowerLevel = commonConfig.DutAverageInputPower - lutResults.PowerOffset;
+            rfsgSession.Initiate();
+            specAnSignal.Dpd.Results.FetchLookupTable(selectorString, commonConfig.Timeout, ref lutResults.LookupTableInputPowers, ref lutResults.LookupTableComplexGains);
+            return lutResults;
+
+        }
+
+        public static MemoryPolynomialResults ApplyPolynomialDPD(RFmxSpecAnMX specAnSignal, NIRfsg rfsgSession, CommonConfiguration commonConfig, MemoryPolynomialConfiguration mpConfig, Waveform waveform, string selectorString = "")
+        {
+            MemoryPolynomialResults mPResults = new MemoryPolynomialResults();
+            IntPtr rfsgPtr;
+            RFmxSpecAnMXDpdApplyDpdIdleDurationPresent idlePresent;
+            idlePresent = waveform.IdleDurationPresent ? RFmxSpecAnMXDpdApplyDpdIdleDurationPresent.True : RFmxSpecAnMXDpdApplyDpdIdleDurationPresent.False;
+            rfsgPtr = rfsgSession.GetInstrumentHandle().DangerousGetHandle();
+            if (mpConfig.IterativeDpdEnabled == RFmxSpecAnMXDpdIterativeDpdEnabled.False)
+            {
+                mpConfig.NumberOfIteration = 1;
+            }
+
+            for (int i = 0; i < mpConfig.NumberOfIteration; i++)
+            {
+                specAnSignal.Dpd.Configuration.ConfigurePreviousDpdPolynomial(selectorString, mPResults.DpdPolynomial);
+                specAnSignal.Initiate("", selectorString);
+                specAnSignal.Dpd.ApplyDpd.ApplyDigitalPredistortion(selectorString, waveform.WaveformData, idlePresent, commonConfig.Timeout, ref mPResults.WaveformWithDpdComplexSingle,
```

Comment:
I would like this instead with my change on 169:

```suggestion
                specAnSignal.Dpd.ApplyDpd.ApplyDigitalPredistortion(selectorString, waveform.WaveformData, idlePresent, timeout_s, ref mPResults.WaveformWithDpdComplexSingle,
```